tools/common.props

```diff
@@ -15,7 +15,5 @@
     <AssemblyOriginatorKeyFile>$(MSBuildThisFileDirectory)Key.snk</AssemblyOriginatorKeyFile>
     <LangVersion>latest</LangVersion>
   </PropertyGroup>
-  <ItemGroup Condition="'$(BuildingInsideVisualStudio)' != 'true'">
-    <PackageReference Include="Microsoft.NETCore.Compilers" Version="$(RoslynVersion)" PrivateAssets="All" />
```

Comment:
Why don't you need this anymore? Do you have a min req for .NET Core that is guaranteed to have C# 7.3?

Reply:
Yes, but it looks like the csc.dll within roslyn that comes with the dotnet cli that we install has a recent enough version and has support for C#7.3.(https://dotnet.myget.org/feed/roslyn/package/nuget/Microsoft.NETCore.Compilers/2.8.1-beta6-62911-06)
![image](https://user-images.githubusercontent.com/6527137/40262487-9ceb1624-5abc-11e8-9c69-63f562db0412.png)

Otherwise, the code change in this PR to remove the cast wouldn't compile (based on recent overload resolution changes - https://github.com/dotnet/roslyn/pull/24756). That change wouldn't build on C#7.2.

> System\Buffers\Writer\BufferWriter_writable.cs(14,16): error CS0306: The type 'Span<byte>' may not be used as a type argument [D:\GitHub\Fork\corefxlab\src\System.Buffers.ReaderWriter\System.Buffers.ReaderWriter.csproj]